src/runtime/components/server/wrapper_heap.cpp

```diff
@@ -319,25 +329,27 @@ namespace hpx { namespace components { namespace detail
 
     bool wrapper_heap::init_pool()
     {
-        HPX_ASSERT(size_ == 0);
         HPX_ASSERT(first_free_ == nullptr);
 
-        std::size_t s = step_ * element_size_; //-V104 //-V707
-        pool_ = allocator_type::alloc(s);
+        std::size_t const total_num_bytes =
+            parameters_.capacity * parameters_.element_size;
+        pool_ = static_cast<char *>(allocator_type::alloc(total_num_bytes));
         if (nullptr == pool_)
         {
             return false;
         }
 
-        first_free_ = pool_;
-        size_ = s / element_size_; //-V104
-        free_size_ = size_;
+        first_free_ = (reinterpret_cast<std::size_t>(pool_)
```

Comment:
uintptr_t maybe?

Reply:
`std::size_t` is good enough as it is guaranteed to be able to hold pointers.